index.src.html

```diff
@@ -1920,33 +1935,36 @@
       may yield decreased performance and power efficiency compared to purely
       software based codecs.
 
-  Given these tradeoffs, a good example of using "require" would be if an author
-  intends to provide their own software based fallback via WebAssembly.
+  Given these tradeoffs, a good example of using "prefer-hardware" would be if
+  an author intends to provide their own software based fallback via
+  WebAssembly.
 
-  Alternatively, a good example of using "disallow" would be if an author is
-  especially sensitive to the higher startup latency or decreased robustness
+  Alternatively, a good example of using "prefer-software" would be if an author
+  is especially sensitive to the higher startup latency or decreased robustness
   generally associated with hardware acceleration.
 </div>
 
 <dl>
-  <dt><dfn enum-value for=HardwareAcceleration>allow</dfn></dt>
+  <dt><dfn enum-value for=HardwareAcceleration>no-preference</dfn></dt>
   <dd>
     Indicates that the User Agent may use hardware acceleration if it is
     available and compatible with other aspects of the codec configuration.
   </dd>
-  <dt><dfn enum-value for=HardwareAcceleration>deny</dfn></dt>
+  <dt><dfn enum-value for=HardwareAcceleration>prefer-software</dfn></dt>
   <dd>
-    Indicates that the User Agent must not use hardware acceleration.
+    Indicates that the User Agent SHOULD prefer a software codec implementation.
+    User Agents may ignore this value for any reason.
 
-    NOTE: This will cause the configuration to be unsupported on platforms where
+    NOTE: This may cause the configuration to be unsupported on platforms where
     an unaccelerated codec is unavailable or is incompatible with other aspects
     of the codec configuration.
   </dd>
-  <dt><dfn enum-value for=HardwareAcceleration>require</dfn></dt>
+  <dt><dfn enum-value for=HardwareAcceleration>prefer-hardware</dfn></dt>
   <dd>
-    Indicates that the User Agent must use hardware acceleration.
+    Indicates that the User Agent SHOULD prefer hardware acceleration. User
+    Agents may ignore this value for any reason.
 
-    NOTE: This will cause the configuration to be unsupported on platforms where
+    NOTE: This may cause the configuration to be unsupported on platforms where
```

Comment:
So this leaves implementations with the ability to mark some configurations as 'unsupported' because of the hint?
This seems to contradict the desire to not allow fingerprinting.
This might create compatibility issues between privacy-aware implementations and privacy-unaware implementations.
I think we should be more restrictive here to ensure leaking fingerprinting is not an option and keep good web compatibility.

Reply:
A few points:
* We must have some sort of configuration rejection mechanism. Your argument applies to every field in the configuration dictionary even if we drop this hunk of text for `hardwareAcceleration`, so it seems impractical. I think this text strikes the best balance we can without becoming prescriptive in codecs (due to registry, literally impossible), resolutions, etc. 
* Cross-UA compatibility isn't a concern in the context of this property, since authors setting `prefer-hardware` or `prefer-software` will already have to be prepared for rejection for many reasons: lack of support, too many codecs in use, gpu disabled, resolution too small or too large, etc,
* You agreed to this here: https://github.com/w3c/webcodecs/issues/239#issuecomment-876667296 and that was what was presented to the WG, who have deferred to the editors consensus.

Reply:
> * Your argument applies to every field in the configuration dictionary even if we drop this hunk of text for `hardwareAcceleration`, so it seems impractical.

The hint is there to sort the codec list, not to reduce the codec list.
As such, the hint should not have allow to change the outcome of supported or not, it should only allow to change the codec behavior if the config is supported.
If a script is able to test the same configuration with two different preferences and in one case, get an unsupported error, we are back to fingerprinting.

As for other configuration field, my understanding is that Media Capabilities should expose the same information.
Or the values make it not supported.
In case this cannot be supported and MC does not disclose that information, WebCodecs is leaking fingerprinting information. We should then do the usual fingerprinting analysis: list the mechanism, see whether it can be avoided...

Looking at video decoder as an example, only optimizeForLatency seems new from Media Capabilities.
It is a hint, and I do not see that it may trigger unsupported configuration.

> * You agreed to this here

I read too quickly the summary, I am sorry to realize that now.
In the last meeting I attended, @chcunningham and I talked about that exact possibility and I expressed my opposition.

> * Cross-UA compatibility isn't a concern in the context of this property

I disagree, cross-UA compatibility usually derives from specs that are not tight enough, which is the case here.

Reply:
> The hint is there to sort the codec list, not to reduce the codec list.

The codec list (keep in mind the configuration is manifold over profiles and resolutions) is already variable in size within the a context of a single-UA regardless of this property, hence my second point. There are no cross-UA compatibility issues that are not also single-UA issues an author would need to be aware of the take advantage of this property.

> In case this cannot be supported and MC does not disclose that information, WebCodecs is leaking fingerprinting information. 

There's no difference in what a given UA reveals between MediaCapabilities and WebCodecs now that this is a hint. Just as Media Capabilities does not define what codecs, resolutions, profiles, or implementations are supported (indeed the very point of the API is to answer that question!) and what it means to be power efficient, so to does the text here allow UA flexibility in making those determinations .

> I disagree, cross-UA compatibility usually derives from specs that are not tight enough, which is the case here.

I think you've misunderstood my comment. While your statement is true in some contexts, it is not true in the context of capabilities that are _standardized_ to vary across UAs and platforms like codecs.

Reply:
I've added a normative note in hope that will resolve your concerns:

"To avoid fingerprinting issues, User Agents SHOULD ensure rejection or
acceptance of a given {{HardwareAcceleration}} preference reveals no additional
information on top of what is revealed by [[media-capabilities]] or other web
exposed capability systems the User Agent may provide."

I'm happy to change that to a MUST if @padenot @aboba want.